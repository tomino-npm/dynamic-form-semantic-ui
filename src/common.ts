import { FormElement, DataSet } from '@tomino/dynamic-form';
import { initCss } from '@tomino/toolbelt';

export type FormControlProps = {
  formControl: FormElement;
  owner: DataSet;
  handlers?: { [index: string]: (...params: any[]) => any };
};

export const { css, renderCss } = initCss('d-form');
