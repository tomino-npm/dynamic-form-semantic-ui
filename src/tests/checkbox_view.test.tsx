import * as React from 'react';
import * as renderer from 'react-test-renderer';

import { JSONSchema, FormModel, config } from '@tomino/dynamic-form';

import { create } from './form_query_data';
import { TestComponent } from './common';

describe('Form', () => {
  const setDirty = jest.fn();

  beforeAll(() => {
    config.setDirty = () => setDirty();
  });
  describe('Checkbox', () => {
    function componentWithData() {
      const schema: JSONSchema = {
        type: 'object',
        properties: {
          agree: {
            type: 'boolean'
          },
          disagree: {
            type: 'boolean'
          },
          must: {
            type: 'boolean',
            errorMessage: 'You must agree to terms and conditions'
          }
        },
        required: ['must']
      };

      const data = { agree: true, disagree: false };

      const form = new FormModel(
        create.form({
          elements: [
            create.formElement({
              row: 0,
              column: 0,
              width: 8,
              control: 'Checkbox',
              label: 'Agree With Terms and Conditions',
              source: 'agree'
            }),
            create.formElement({
              row: 1,
              column: 0,
              width: 8,
              control: 'Checkbox',
              controlProps: {
                toggle: true
              },
              label: 'Disagree With Terms and Conditions',
              source: 'disagree'
            }),
            create.formElement({
              row: 2,
              column: 0,
              width: 8,
              control: 'Checkbox',
              label: 'Must Agree With Terms and Conditions',
              source: 'must'
            })
          ]
        }),
        schema,
        data
      );

      form.validateWithReport();

      // just another notation
      return <TestComponent form={form} />;
    }

    it('renders correctly', () => {
      const component = renderer.create(componentWithData());
      expect(component).toMatchSnapshot();
    });

    it('changes value and all related formulas', () => {
      const component = renderer.create(componentWithData());
      const root = component.root;
      const agree = root.findByProps({ name: 'agree' });
      agree.props.onChange(null, { checked: false });

      const disagree = root.findByProps({ name: 'disagree' });
      disagree.props.onChange(null, { checked: true });

      const must = root.findByProps({ name: 'must' });
      must.props.onChange(null, { checked: true });
      expect(component).toMatchSnapshot();
      expect(setDirty).toBeCalled();
    });

    return { componentWithData };
  });
});
