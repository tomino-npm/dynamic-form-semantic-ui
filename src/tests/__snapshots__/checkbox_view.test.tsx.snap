// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form Checkbox changes value and all related formulas 1`] = `
<div
  className="ui segment ui form"
>
  <style>
    .d-form-bgcibidbjc { 
  color: pink;
  font-size: 100px !important;
 }
.d-form-Aebfdiadfh { 
  padding: 9px 0px;
 }
.d-form-Adiagibjfa { 
  margin-right: 12px;
  margin-bottom: 6px;
 }
.d-form-bidfabidci { 
  margin: 6px 0px 12px 0px;
 }
.d-form-Abdgbhfjheg { 
  margin-bottom: 0px !important;
 }

  .fieldSet-bejaheefjb {
    border: solid 1px #ddd;
    border-radius: 5px;
    background-color: #fafafa;
  }


  .fieldSet-bejaheefjb legend {
    padding: 0px 6px;
    font-weight: bold;
  }
  </style>
  <div
    className="ui form"
  >
    <div
      className="fields"
    >
      <div
        className="eight wide field"
      >
        <div
          className="field"
        >
          <div
            className="ui checkbox d-form-bgcibidbjc"
            onChange={[Function]}
            onClick={[Function]}
            onMouseDown={[Function]}
          >
            <input
              checked={false}
              className="hidden"
              name="agree"
              onClick={[Function]}
              readOnly={true}
              tabIndex={0}
              type="checkbox"
            />
            <label>
              Agree With Terms and Conditions
            </label>
          </div>
        </div>
      </div>
    </div>
    <div
      className="fields"
    >
      <div
        className="eight wide field"
      >
        <div
          className="field"
        >
          <div
            className="ui checked toggle checkbox d-form-bgcibidbjc"
            onChange={[Function]}
            onClick={[Function]}
            onMouseDown={[Function]}
          >
            <input
              checked={true}
              className="hidden"
              name="disagree"
              onClick={[Function]}
              readOnly={true}
              tabIndex={0}
              type="checkbox"
            />
            <label>
              Disagree With Terms and Conditions
            </label>
          </div>
        </div>
      </div>
    </div>
    <div
      className="fields"
    >
      <div
        className="eight wide field"
      >
        <div
          className="field"
        >
          <div
            className="ui checked checkbox d-form-bgcibidbjc"
            onChange={[Function]}
            onClick={[Function]}
            onMouseDown={[Function]}
          >
            <input
              checked={true}
              className="hidden"
              name="must"
              onClick={[Function]}
              readOnly={true}
              tabIndex={0}
              type="checkbox"
            />
            <label>
              Must Agree With Terms and Conditions
            </label>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Form Checkbox renders correctly 1`] = `
<div
  className="ui segment ui form"
>
  <style>
    .d-form-bgcibidbjc { 
  color: pink;
  font-size: 100px !important;
 }
.d-form-Aebfdiadfh { 
  padding: 9px 0px;
 }
.d-form-Adiagibjfa { 
  margin-right: 12px;
  margin-bottom: 6px;
 }
.d-form-bidfabidci { 
  margin: 6px 0px 12px 0px;
 }
.d-form-Abdgbhfjheg { 
  margin-bottom: 0px !important;
 }

  .fieldSet-bejaheefjb {
    border: solid 1px #ddd;
    border-radius: 5px;
    background-color: #fafafa;
  }


  .fieldSet-bejaheefjb legend {
    padding: 0px 6px;
    font-weight: bold;
  }
  </style>
  <div
    className="ui form"
  >
    <div
      className="fields"
    >
      <div
        className="eight wide field"
      >
        <div
          className="field"
        >
          <div
            className="ui checked checkbox d-form-bgcibidbjc"
            onChange={[Function]}
            onClick={[Function]}
            onMouseDown={[Function]}
          >
            <input
              checked={true}
              className="hidden"
              name="agree"
              onClick={[Function]}
              readOnly={true}
              tabIndex={0}
              type="checkbox"
            />
            <label>
              Agree With Terms and Conditions
            </label>
          </div>
        </div>
      </div>
    </div>
    <div
      className="fields"
    >
      <div
        className="eight wide field"
      >
        <div
          className="field"
        >
          <div
            className="ui toggle checkbox d-form-bgcibidbjc"
            onChange={[Function]}
            onClick={[Function]}
            onMouseDown={[Function]}
          >
            <input
              checked={false}
              className="hidden"
              name="disagree"
              onClick={[Function]}
              readOnly={true}
              tabIndex={0}
              type="checkbox"
            />
            <label>
              Disagree With Terms and Conditions
            </label>
          </div>
        </div>
      </div>
    </div>
    <div
      className="fields"
    >
      <div
        className="eight wide field"
      >
        <div
          className="field"
        >
          <div
            className="ui checkbox d-form-bgcibidbjc"
            onChange={[Function]}
            onClick={[Function]}
            onMouseDown={[Function]}
          >
            <input
              checked={false}
              className="hidden"
              name="must"
              onClick={[Function]}
              readOnly={true}
              tabIndex={0}
              type="checkbox"
            />
            <label>
              Must Agree With Terms and Conditions
            </label>
          </div>
          <div>
            <div
              className="ui red pointing label"
              onClick={[Function]}
            >
              You must agree to terms and conditions
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
